### ISSUE TITLE
Ubuntu 18 doesn't work any GUI apps.

### ISSUE THREAD

**Issue by abikonya** [microsoft/WSL#8493]
Ubuntu 18 doesn't work any GUI apps.
Before update all apps works with workaround "ln -s /mnt/wslg/.X11-unix /tmp/"

_Originally posted by @abikonya in https://github.com/microsoft/WSL/discussions/8487#discussioncomment-2904830_

**Comment by elsaco** [comment]
Sample screenshot running emacs on Ubuntu 18.04:
![wslg_ubuntu1804_emacs](https://user-images.githubusercontent.com/3933920/172898372-85e4fe23-9571-48e4-b1ac-470509701caa.png)

Notice the `/mnt/wslg/.X11-unix` directory.

**Comment by abikonya** [comment]
It's very strange I don't have X0 file
![screenshot 2022-06-13 170403](https://user-images.githubusercontent.com/36229338/173330531-d5f35d19-5e67-4ae4-95b2-d7621783b3c3.png)

**Comment by facboy** [comment]
i'm also missing the `X0` file, but i am on 22.04.  in my case it seems that something in `systemd` is either deleting it or interfering with it, as if i disable `systemd` then it is there.

**Comment by benhillis** [comment]
Yes, that's exactly what's happening. Systemd is deleting the temp directory.
